daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -111,6 +114,29 @@ func New(ctx logger.Context) (logger.Logger, error) {
 	return containerStream, nil
 }
 
+// Extract the logStreamName, rendering the template if appropriate
+func getStreamName(ctx logger.Context) (string, error) {
+	if ctx.Config[logStreamKey] != "" {
+		return ctx.Config[logStreamKey], nil
+	} else if ctx.Config[logStreamTemplateKey] != "" {
+		var name string
+		t, err := template.New("t").Parse(ctx.Config[logStreamTemplateKey])
+		if err == nil {
+			buff := bytes.NewBuffer(nil)
+			err = t.Execute(buff, ctx)
+			name = buff.String()
+		}
+		if err == nil && strings.IndexAny(name, ":*") >= 0 {
+			// Arguably, we could let AWS reject this later, but by then your container is
```

Comment:
`(*logStream).create()` synchronously calls `CreateLogStream` which should reject invalid names.  Since errors resulting from that call cause `New()` to return an error, the container should not actually start running.  However, I don't object to local validity checks in the name like this.